### ISSUE TITLE
Use named tuple in IProviderState.GetSnapshotAsync method return type

### ISSUE THREAD

**Issue by tomliversidge** [AsynkronIT/protoactor-dotnet#116]
Currently the GetSnapshotAsync returns a `Task<Tuple<object, long>>` but it would be nicer to use the new named tuple support in C#7 - this will avoid the ugly Item1 and Item2 syntax. 

For example, could have
`Task<(object state, long index)> GetSnapshotAsync`

and call it like
`var (state, index) = await providerState.GetSnapshotAsync()`

or

`var result = await providerState.GetSnapshotAsync()`
`_state = result.state` 

Might be better to do this soon if wanted before more persistence plugins are developed.

**Comment by raskolnikoov** [comment]
Yes this I like :)

**Comment by cpx86** [comment]
Absolutely yes :)

**Comment by raskolnikoov** [comment]
Close this now that we have PR https://github.com/AsynkronIT/protoactor-dotnet/pull/117 implementing this?

**Comment by tomliversidge** [comment]
You should be able to reference the issue in the commit message to close it automatically when it is merged: https://help.github.com/articles/closing-issues-via-commit-messages/

TLDR: just add `Fixes #116` to the commit :)
